pytest_splunk_addon/splunk.py

```diff
@@ -416,32 +423,12 @@ def splunk(request, file_system_prerequisite):
     """
     splunk_type = request.config.getoption("splunk_type")
     LOGGER.info("Get the Splunk instance of splunk_type=%s", splunk_type)
-    if splunk_type == "external":
-        request.fixturenames.append("splunk_external")
-        splunk_info = request.getfixturevalue("splunk_external")
-    elif splunk_type == "docker":
-        os.environ["SPLUNK_APP_PACKAGE"] = request.config.getoption("splunk_app")
-        try:
-            config = configparser.ConfigParser()
-            config.read(
-                os.path.join(
-                    request.config.getoption("splunk_app"),
-                    "default",
-                    "app.conf",
-                )
-            )
-            os.environ["SPLUNK_APP_ID"] = config["package"]["id"]
-        except Exception:
-            os.environ["SPLUNK_APP_ID"] = "TA_package"
-        os.environ["SPLUNK_HEC_TOKEN"] = request.config.getoption("splunk_hec_token")
-        os.environ["SPLUNK_USER"] = request.config.getoption("splunk_user")
-        os.environ["SPLUNK_PASSWORD"] = request.config.getoption("splunk_password")
-        os.environ["SPLUNK_VERSION"] = request.config.getoption("splunk_version")
-
-        request.fixturenames.append("splunk_docker")
-        splunk_info = request.getfixturevalue("splunk_docker")
-    else:
-        raise Exception
+    splunk_fixture = f"splunk_{splunk_type}"
+    try:
+        request.fixturenames.append(splunk_fixture)
+        splunk_info = request.getfixturevalue(splunk_fixture)
+    except Exception as e:
```

Comment:
Ideally it should have a simple unit test covering this logic but I am fine moving forward with this PR.